Neos.RedirectHandler.DatabaseStorage/Classes/RedirectionStorage.php

```diff
@@ -0,0 +1,231 @@
+<?php
+namespace Neos\RedirectHandler\DatabaseStorage;
+
+/*
+ * This file is part of the Neos.RedirectHandler package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\ORM\OptimisticLockException;
+use Neos\RedirectHandler\DatabaseStorage\Domain\Model\Redirection;
+use Neos\RedirectHandler\DatabaseStorage\Domain\Repository\RedirectionRepository;
+use Neos\RedirectHandler\Exception;
+use Neos\RedirectHandler\Redirection as RedirectionDto;
+use Neos\RedirectHandler\Service\SettingsService;
+use Neos\RedirectHandler\Storage\RedirectionStorageInterface;
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Flow\Mvc\Routing\RouterCachingService;
+use TYPO3\Flow\Persistence\PersistenceManagerInterface;
+
+/**
+ * Database Storage for the Redirections
+ *
+ * @Flow\Scope("singleton")
+ */
+class RedirectionStorage implements RedirectionStorageInterface
```

Comment:
I am against the duplication, Neos\RedirectHandler\DatabaseStorage\RedirectionStorage is already pretty much duplicate everything. Custom "use" statements are the solution if you feel it is ambigous in your code. IMHO the FQCN should make sense, everything else is to be solved in code using it.